frontend/src/Components/Modal.js

```diff
@@ -0,0 +1,56 @@
+import ReactDom from 'react-dom';
+import { useRef } from 'react';
+import '../styles/components/Modal.css';
+const Modal = ({ heading, children, active, setActive }) => {
+  const modalRef = useRef();
+  const closeModal = (e) => {
+    e.stopPropagation();
+    setActive(false);
+  };
+  const toggleOn = () => {
+    modalRef.current.style.display = 'flex';
+  };
+  const toggleOn1 = () => {
+    modalRef.current.style.opacity = '1';
+  };
+  const toggleOff = () => {
+    if (modalRef.current) {
+      modalRef.current.style.opacity = '0';
+    }
+  };
+  const toggleOff1 = () => {
+    if (modalRef.current) {
+      modalRef.current.style.display = 'none';
+    }
+  };
+  return ReactDom.createPortal(
+    <div className={`modal-backdrop`} ref={modalRef}>
+      <div style={{ display: 'none' }}>
+        {active ? toggleOn() : toggleOff()}
```

Comment:
Do you think using a react animation library such as Framer Motion could help prevent needing to add setTimeouts inside the code and directly change inline styles?